atom/common/api/resources/windows_bindings.js

```diff
@@ -1,53 +1,104 @@
+var binding = require('binding').Binding.create('windows')
+
 var ipc = require('ipc_utils')
 
 var id = 1;
 
-var binding = {
-  getCurrent: function () {
-    var cb, getInfo;
-    if (arguments.length == 1) {
+binding.registerCustomHook(function (bindingsAPI, extensionId) {
+  var apiFunctions = bindingsAPI.apiFunctions
+  var windows = bindingsAPI.compiledApi
+
+  apiFunctions.setHandleRequest('getLastFocused', function (getInfo, cb) {
+    console.warn('chrome.windows.getLastFocused is not supported yet')
+  })
+
+  apiFunctions.setHandleRequest('remove', function (windowId, cb) {
+    console.warn('chrome.windows.remove is not supported yet')
+  })
+
+  apiFunctions.setHandleRequest('get', function (windowId, getInfo, cb) {
+    console.warn('chrome.windows.get is not supported yet')
+  })
+
+  apiFunctions.setHandleRequest('create', function (createData, cb) {
+    console.warn('chrome.windows.create is not supported yet')
+  })
+
+  apiFunctions.setHandleRequest('getCurrent', function () {
+    var cb, getInfo
```

Comment:
`setHandleRequest` is supposed to normalize these so you always get two arguments even if the user only supplies one. I think this code should still work ok, but the argument length condition is no longer necessary